src/libs/filters/camera.c

```diff
@@ -0,0 +1,363 @@
+/*
+    This file is part of darktable,
+    Copyright (C) 2023 darktable developers.
+
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*
+  This file contains the necessary routines to implement a filter for the filtering module
+*/
+
+typedef struct _widgets_camera_t
+{
+  dt_lib_filtering_rule_t *rule;
+
+  GtkWidget *name;
+  GtkWidget *pop;
+  GtkWidget *name_tree;
+  gboolean tree_ok;
+  int internal_change;
+} _widgets_camera_t;
+
+static void _camera_synchronise(_widgets_camera_t *source)
+{
+  _widgets_camera_t *dest = NULL;
+  if(source == source->rule->w_specific_top)
+    dest = source->rule->w_specific;
+  else
+    dest = source->rule->w_specific_top;
+
+  if(dest)
+  {
+    source->rule->manual_widget_set++;
+    const gchar *txt1 = gtk_entry_get_text(GTK_ENTRY(source->name));
+    gtk_entry_set_text(GTK_ENTRY(dest->name), txt1);
+    source->rule->manual_widget_set--;
+  }
+}
+
+static void _camera_changed(GtkWidget *widget,
+                            gpointer user_data)
+{
+  _widgets_camera_t *camera = (_widgets_camera_t *)user_data;
+  if(camera->rule->manual_widget_set) return;
+
+  _rule_set_raw_text(camera->rule, gtk_entry_get_text(GTK_ENTRY(camera->name)), TRUE);
+  _camera_synchronise(camera);
+}
+
+static gboolean _camera_focus_out(GtkWidget *entry,
+                                  GdkEventFocus *event,
+                                  gpointer user_data)
+{
+  _widgets_camera_t *camera = (_widgets_camera_t *)user_data;
+  if(camera->rule->cleaning) return FALSE;
+  _camera_changed(entry, user_data);
+  return FALSE;
+}
+
+void _camera_tree_update(_widgets_camera_t *camera)
+{
+  dt_lib_filtering_t *d = camera->rule->lib;
+
+  char query[1024] = { 0 };
+
+  GtkTreeIter iter;
+  GtkTreeModel *name_model = gtk_tree_view_get_model(GTK_TREE_VIEW(camera->name_tree));
+  gtk_list_store_clear(GTK_LIST_STORE(name_model));
+
+  // clang-format off
+  g_snprintf(query, sizeof(query),
+             "SELECT cm.name, COUNT(*) AS count"
+             " FROM main.images AS mi, main.cameras AS cm"
+             " WHERE mi.camera_id = cm.id AND %s"
+             " GROUP BY mi.camera_id",
```

Comment:
Why not 

```
SELECT cm.name AS camera, COUNT(*) AS count
...
GROUP BY camera 
ORDER BY camera
```

This would order the list of camera, which seems better to me.